### PR TITLE
Issue 6 I should be able to send a mail randomly using gatling

### PR DESCRIPTION
**Pull request by rouazana**
Follow #14

### CODE REVIEW

File: src/test/scala-2.11/org/apache/james/gatling/jmap/scenari/JmapSendMessagesScenario.scala [hunk 1]
```diff
@@ -6,17 +6,19 @@ import org.apache.james.gatling.jmap.scenari.common.CommonSteps
 import org.apache.james.gatling.jmap.{HttpSettings, JmapMessages}
 
 import scala.concurrent.duration._
+import org.apache.james.gatling.control.{Username, Password, User}
 
 class JmapSendMessagesScenario extends Simulation {
 
   val userCount = 100
-  val feeder = UserFeeder.createUserFeederWithInboxAndOutbox(userCount)
+  val users = UserFeeder.createCompletedUserFeederWithInboxAndOutbox(userCount)
```
Comment (mbaechler) [inline src/test/scala-2.11/org/apache/james/gatling/jmap/scenari/JmapSendMessagesScenario.scala]: what if we create a function that return the array for users and we map it to a feeder when we need it ?
It sounds more logical, don't you think ?

Comment (rouazana) [reply src/test/scala-2.11/org/apache/james/gatling/jmap/scenari/JmapSendMessagesScenario.scala]: I tried something, WDYT?
At least the code seems cleaner.

File: src/main/scala-2.11/org/apache/james/gatling/control/UserCreator.scala [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    JamesWebAdministration.addDomain(domain).get
+
+    generateUsers(userCount, domain)
+      .map(JamesWebAdministration.addUser)
+  }
+
+  private def generateUsers(userCount: Int, domain: Domain): List[User] =
+    (0 until userCount)
+      .map(i => User.random(domain))
+      .toList
```
Comment (mbaechler) [inline src/main/scala-2.11/org/apache/james/gatling/control/UserCreator.scala]: why do you need a list ?

File: src/main/scala-2.11/org/apache/james/gatling/control/UserCreator.scala [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+  def createUsersWithInboxAndOutbox(userCount: Int): List[Future[User]] =
+        createUsers(userCount)
+          .map(userFuture => userFuture.andThen {
+            case Success(user) => registerSystemMailboxes(user)
+          })
+
+  private def createUsers(userCount: Int): List[Future[User]] = {
+    val domain = Domain.random
+    JamesWebAdministration.addDomain(domain).get
```
Comment (mbaechler) [inline src/main/scala-2.11/org/apache/james/gatling/control/UserCreator.scala:21]: what about doing a map here ?

File: src/test/scala-2.11/org/apache/james/gatling/jmap/scenari/JmapSendMessagesScenario.scala [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+import org.apache.james.gatling.jmap.scenari.common.CommonSteps
+import org.apache.james.gatling.jmap.{HttpSettings, JmapMessages}
+
+import scala.concurrent.duration._
+import org.apache.james.gatling.control.{Username, Password, User}
+
+class JmapSendMessagesScenario extends Simulation {
+
+  val userCount = 100
+  val users = UserFeeder.createCompletedUserFeederWithInboxAndOutbox(userCount)
```
Comment (mbaechler) [inline src/test/scala-2.11/org/apache/james/gatling/jmap/scenari/JmapSendMessagesScenario.scala]: you no longer need the feeder here, don't you ?

Comment (rouazana) [reply src/test/scala-2.11/org/apache/james/gatling/jmap/scenari/JmapSendMessagesScenario.scala]: yes, I was working on it

File: src/main/scala-2.11/org/apache/james/gatling/control/UserCreator.scala (hunk 2 above)
Comment (rouazana) [reply src/main/scala-2.11/org/apache/james/gatling/control/UserCreator.scala]: replaced (everywhere) by a Seq

File: src/main/scala-2.11/org/apache/james/gatling/control/UserCreator.scala (hunk 3 above)
Comment (rouazana) [reply src/main/scala-2.11/org/apache/james/gatling/control/UserCreator.scala:21]: and returning Future[List[Future[User]]? That seems to much future for me :)

**Review by mbaechler — APPROVED** [review]

**Review by chibenwa — APPROVED** [review]

**Pull request merged**